### PR TITLE
[online-cmaq] Add point sources changes needed for workflow

### PR DESCRIPTION
**Pull request by bbakernoaa**
## DESCRIPTION OF CHANGES: 

This adds the necessary changes for the point source changes, particularly the `parm/aqm.rc` and the `scripts/exregional_point_source.sh`

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## TESTS CONDUCTED: 

- [x] hera.intel
- [ ] orion.intel
- [ ] cheyenne.intel
- [ ] cheyenne.gnu
- [ ] gaea.intel
- [ ] jet.intel
- [ ] wcoss2.intel
- [ ] NOAA Cloud (indicate which platform)
- [ ] Jenkins
- [ ] fundamental test suite
- [ ] comprehensive tests (specify *which* if a subset was used)

## ISSUE: 

Related to https://github.com/NOAA-EMC/AQM/issues/48

## CHECKLIST

- [x] My code follows the style guidelines in the Contributor's Guide
- [x] I have performed a self-review of my own code using the Code Reviewer's Guide
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [ ] My changes do not require updates to the documentation (explain).
- [x] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

## LABELS (optional): 

A Code Manager needs to add the following labels to this PR: 
- [ ] Work In Progress
- [ ] bug
- [ ] enhancement
- [ ] documentation
- [ ] release
- [ ] high priority
- [ ] run_ci
- [ ] run_we2e_fundamental_tests
- [ ] run_we2e_comprehensive_tests
- [ ] Needs Cheyenne test 
- [ ] Needs Jet test 
- [ ] Needs Hera test 
- [ ] Needs Orion test 
- [ ] help wanted

### CODE REVIEW

**Comment by chan-hoo** [comment]
@bbakernoaa, I think we should update the hash of the ufs-weather-model with the latest one. Is this correct?

**Comment by chan-hoo** [comment]
@bbakernoaa, can you update the hashes of 'arl_nexus' and 'aqm-utils' with '3842818' and '81a99ca', respectively? Can you merge the latest 'online-cmaq' into your branch for this PR too?

File: scripts/exregional_point_source.sh [hunk 1]
```diff
@@ -162,7 +164,7 @@ POST_STEP
 #
 #-----------------------------------------------------------------------
 #
-mv_vrfy "${DATA}/PT" ${INPUT_DATA}
+#mv_vrfy "${DATA}/PT" ${INPUT_DATA}
```
Comment (chan-hoo) [inline scripts/exregional_point_source.sh]: @bbakernoaa, if this part and above, which were commented out, are not necessary, please remove them from the script.

**Comment by chan-hoo** [comment]
@bbakernoaa, can you merge the latest 'online-cmaq' branch to your branch for this PR and update the hash of the ufs weather model in `Externals_AQM.cfg`? Then I'll test this PR.

File: Externals.cfg [hunk 2]
```diff
@@ -12,7 +12,7 @@ protocol = git
 repo_url = https://github.com/ufs-community/ufs-weather-model
 # Specify either a branch name or a hash but not both.
 #branch = develop
-hash = 3610672
+hash = de1546f
 local_path = sorc/ufs-weather-model
```
Comment (chan-hoo) [inline Externals.cfg]: @bbakernoaa, I changed the structure of the 'Externals.cfg' files to integrate online-cmaq into the develop branch. The basic concept is that we maintain a separate Externals.cfg file named 'Externals_AQM.cfg' for online-cmaq and we don't change any hashes in 'Externals.cfg'. Therefore, please change this hash to the original and change the hash of the ufs-weather-model in `Externals_AQM.cfg`. I think you didn't merge the latest online-cmaq branch into your branch yet becasue I moved `Externals_AQM.cfg` from 'externals' to the parent directory.

**Comment by chan-hoo** [comment]
@bbakernoaa, can you remove the conflict by merging the latest 'online-cmaq' branch into your PR branch?

**Comment by bbakernoaa** [comment]
done @chan-hoo

**Comment by chan-hoo** [comment]
@bbakernoaa, please change the hash of the ufs weather model in `Externals_AQM.cfg` (not in `Externals.cfg`).

**Comment by chan-hoo** [comment]
Thanks. I'll run the sample case.

**Comment by chan-hoo** [comment]
@bbakernoaa, can you change the permission of the file `scripts/exregional_point_source.sh`? -rw-r--r--  1 chan-hoo.jeon lam  4772 Jan 18 18:54 exregional_point_source.sh (chmod +x exregional_point_source.sh)

**Comment by chan-hoo** [comment]
@bbakernoaa, I got the following error:   ```ln: failed to create hard link '/lfs/h2/emc/lam/save/chan-hoo.jeon/online-cmaq_test/expt_dirs/../nco_dirs/com/aqm/v7.0/aqm_test_v16.20230116/18/INPUT/PT.nc' => '/lfs/h2/emc/lam/save/chan-hoo.jeon/online-cmaq_test/expt_dirs/../nco_dirs/tmp/point_source.28612133.cbqs01/tmp_PT_SOURCE/pt-2023011618.nc': No such file or directory```.

**Comment by bbakernoaa** [comment]
I'm looking at it now.

**Comment by chan-hoo** [comment]
It should be `ln_vrfy -sf ${INPUT_DATA}/INPUT/PT.nc ${DATA}/pt-${yyyymmddhh}.nc`.

**Comment by chan-hoo** [comment]
Can you change 'ln_vrfy' to 'mv_vrfy' : `mv_vrfy ${DATA}/pt-${yyyymmddhh}.nc ${INPUT_DATA}/INPUT/PT.nc`

**Comment by chan-hoo** [comment]
I'll merge this PR now and correct any other issues on my end.

**Review by chan-hoo — APPROVED** [review]

**Comment by bbakernoaa** [comment]
@chan-hoo Thanks!

**Comment by drnimbusrain** [comment]
@bbakernoaa @chan-hoo @ytangnoaa @zmoon

I have just tried to do a test 1-day run of the latest ufs-community UFS-SRW-App [online-cmaq] with inclusion of these new point source workflow on WCOSS2.   However, I get an error during the run_fcst that it cannot find the input point source file:

20230120 221001.488 ERROR            PET0000 aqmio.F90:446        - **Error opening NetCDF data set: INPUT/PT.nc** (strerr=No such file or directory, ncerrToCheck=2)

You can see my workflow directory and run directory below:
/lfs/h2/oar/arl/noscrub/Patrick.C.Campbell/run_ufs-srw-app_online-cmaq/run_na_test_ufs-srw-app-ufs-comm-base_20230120
/lfs/h2/oar/ptmp/Patrick.C.Campbell/expt_dirs/aqm_cold_aqmna13_1day_gfsv16_ufs-comm-base_no-canopy-no-fengsha

**Comment by drnimbusrain** [comment]
@zmoon  I will note that I do not see this same issue in our fork of [online-cmaq] UFS-SRW-App.  It seems like we have not synced these PT source workflow changes to our fork.

**Comment by zmoon** [comment]
> we have not synced these PT source workflow changes to our fork.

Yeah that's correct. Was waiting a bit before merging again.

**Pull request merged**